src/vs/base/browser/ui/list/listWidget.ts

```diff
@@ -527,11 +527,27 @@ class TypeNavigationController<T> implements IDisposable {
 			const label = this.keyboardNavigationLabelProvider.getKeyboardNavigationLabel(this.view.element(index));
 			const labelStr = label && label.toString();
 
-			if (typeof labelStr === 'undefined' || matchesPrefix(word, labelStr)) {
-				this.previouslyFocused = start;
-				this.list.setFocus([index]);
-				this.list.reveal(index);
-				return;
+			if (this.list.options.typeNavigationEnabled) {
+				if (typeof labelStr !== 'undefined') {
+					const prefix = matchesPrefix(word, labelStr);
+					const fuzzy = matchesFuzzy2(word, labelStr);
+
+					// ensures that when fuzzy matching, it doesn't clash with prefix matching (1 input vs 1+ should be prefix and fuzzy respecitvely)
+					const fuzzyScore = fuzzy ? fuzzy[0].end - fuzzy[0].start : 0;
+					if (prefix || fuzzyScore > 1) {
+						this.previouslyFocused = start;
+						this.list.setFocus([index]);
+						this.list.reveal(index);
+						return;
+					}
+				}
+			} else {
+				if (typeof labelStr === 'undefined' || matchesPrefix(word, labelStr)) {
```

Comment:
Very strange to see ` else { if (`. Also, inside this `else`, we're sure  `typeof labelStr === 'undefined'`, so we don't need to check for that.